src/Data/Nat/Binary/Subtraction.agda

```diff
@@ -0,0 +1,344 @@
+    1ᵇ + 2x                               ≡⟨ sym (suc≗1+ 2x) ⟩
+    suc 2x                                ≡⟨ sym (1+[2_]-suc-double x) ⟩
+    x'                                    ∎
+  where
+  open ≤-Reasoning
+  x∸y = x ∸ y;  2x = double x;  2y = double y;  2[x∸y] = double x∸y;  x∸y>0 = x<y⇒0<y∸x y<x
+  x∸y≢0       = >⇒≢ x∸y>0
+  2[x∸y]≢0    = x≢0⇒double[x]≢0 x∸y≢0
+  2[suc[y]]≢0 = x≢0⇒double[x]≢0 (suc≢0 {y})
+x≥y⇒[x∸y]+y≡x {x'@(1+[2 x ])} {y'@(1+[2 y ])} (inj₁ (odd<odd y<x)) =
+  begin-equality
+    (x' ∸ y') + y'                 ≡⟨ cong ((x' ∸ y') +_) (1+[2_]-suc-double y) ⟩
+    double x∸y + (suc (double y))  ≡⟨ x+suc[y]≡suc[x]+y (double x∸y) (double y) ⟩
+    suc (double x∸y) + double y    ≡⟨ suc-+ (double x∸y) (double y) ⟩
+    suc (double x∸y + double y)    ≡⟨ cong suc $ sym $ double-distrib-+ x∸y y ⟩
+    suc (double (x∸y + y))         ≡⟨ sym (1+[2_]-suc-double _) ⟩
+    1+[2 (x∸y + y) ]               ≡⟨ cong 1+[2_] $ x≥y⇒[x∸y]+y≡x (inj₁ y<x) ⟩
+    1+[2 x ]                       ∎
+  where open ≤-Reasoning;  x∸y = x ∸ y
+x≥y⇒[x∸y]+y≡x {x'@(1+[2 x ])} {1+[2 x ]} (inj₂ refl) = begin-equality
+  (x' ∸ x') + x'  ≡⟨ cong (_+ x') (x∸x≡0 x') ⟩
+  0ᵇ + x'         ≡⟨ sym (+-identityˡ x') ⟩
+  x'              ∎
+  where open ≤-Reasoning
+
+x+y∸y≡x : ∀ x y → (x + y) ∸ y ≡ x
+x+y∸y≡x x y = +-cancelʳ-≡ d x d+y≡x+y
+  where
+  d = (x + y) ∸ y;  y≤x+y = x≤y+x y x;  d+y≡x+y = x≥y⇒[x∸y]+y≡x y≤x+y
+
+[x+y]∸x≡y : ∀ x y → (x + y) ∸ x ≡ y
+[x+y]∸x≡y x y = trans (cong (_∸ x) (+-comm x y)) (x+y∸y≡x y x)
+
+∸-identityʳ : RightIdentity 0ᵇ _∸_
+∸-identityʳ = x∸0≡x
+
+x+[y∸x]≡y : ∀ {x y} → x ≤ y → x + (y ∸ x) ≡ y
+x+[y∸x]≡y {x} {y} x≤y = begin-equality
+  x + (y ∸ x)   ≡⟨ +-comm x _ ⟩
+  (y ∸ x) + x   ≡⟨ x≥y⇒[x∸y]+y≡x x≤y ⟩
+  y             ∎
+  where open ≤-Reasoning
+
+[x∸y]+y≡x : ∀ {x y} → y ≤ x → (x ∸ y) + y ≡ x
+[x∸y]+y≡x {x} {y} y≤x = trans (+-comm (x ∸ y) y) (x+[y∸x]≡y y≤x)
+
+toℕ-homo-∸ : ∀ x y → toℕ (x ∸ y) ≡ (toℕ x) ℕ.∸ (toℕ y)
+toℕ-homo-∸ x y with x ≤? y
+... | yes x≤y = begin
+  toℕ (x ∸ y)   ≡⟨ cong toℕ (x≤y⇒x∸y≡0 x≤y) ⟩
+  toℕ 0ᵇ        ≡⟨⟩
+  0             ≡⟨ sym (ℕₚ.m≤n⇒m∸n≡0 m≤n) ⟩
+  m ℕ.∸ n       ∎
+  where
+  open ≡-Reasoning;  m = toℕ x;  n = toℕ y;  m≤n = toℕ-mono-≤ x≤y
+
+... | no x≰y = begin
+  toℕ (x ∸ y)                    ≡⟨ sym (ℕₚ.m+n∸n≡m (toℕ (x ∸ y)) n) ⟩
+  ((toℕ (x ∸ y)) ℕ.+ n) ℕ.∸ n    ≡⟨ cong (ℕ._∸ n) eq ⟩
+  ((m ℕ.∸ n) ℕ.+ n) ℕ.∸ n        ≡⟨ ℕₚ.m+n∸n≡m (m ℕ.∸ n) n ⟩
+  m ℕ.∸ n                        ∎
+  where
+  open ≡-Reasoning
+  m  = toℕ x;   n = toℕ y;   y≤x = <⇒≤ (≰⇒> x≰y);   n≤m = toℕ-mono-≤ y≤x
+  eq = begin
+    toℕ (x ∸ y) ℕ.+ n    ≡⟨ sym (toℕ-homo-+ (x ∸ y) y) ⟩
+    toℕ ((x ∸ y) + y)    ≡⟨ cong toℕ (+-comm (x ∸ y) y) ⟩
+    toℕ (y + (x ∸ y))    ≡⟨ cong toℕ (x+[y∸x]≡y y≤x) ⟩
+    m                    ≡⟨ sym (ℕₚ.m+[n∸m]≡n n≤m) ⟩
+    n ℕ.+ (m ℕ.∸ n)      ≡⟨ ℕₚ.+-comm n (m ℕ.∸ n) ⟩
+    (m ℕ.∸ n) ℕ.+ n      ∎
+
+∸-magma : Magma 0ℓ 0ℓ
+∸-magma = magma _∸_
+
+import Algebra.Morphism.TwoMagmas ∸-magma ℕₚ.∸-magma
```

Comment:
Import statements should go at the top of the file.

Reply:
I set ``import Algebra.Morphism.Consequences`` to the head imports,                         
and change the subject line to                                                              
```                                                                                         
module Magmas∸ℕᵇ-ℕ = Algebra.Morphism.Consequences ∸-magma ℕₚ.∸-magma                       
```                                                                                         
(assuming that you require the "Consequences" renaming below).